### DISCUSSION TITLE
One of two classes not working on Vercel

### DISCUSSION THREAD

**Discussion by carstenjaksch** [tailwindlabs/tailwindcss#12443] [Help] [answered]
Hi,

I have a Skeleton (SvelteKit) project that fetches some classes from an API. This is the affected code:

```svelte
<script lang="ts">
    // SomeComponent.svelte
    export let data: {
        // ...
        bg?: string; // Passed from API: bg-gray-50
        py?: string; // Passed from API: py-4
    };
    let { bg = 'bg-transparent', py = 'py-0' } = data; // defaults

    let classes = '';
    export { classes as class };
    classes += ` ${bg} ${py}`;

    // Later px-[--container-px] gets conditionally assigned to classes and works as expected.
</script>

<div class="some-component {classes}">
    <!-- ... -->
</div>
```

The weird thing is, the `py` class works, but not the `bg` class (final HTML has both, but not the styles of `bg`). And only on Vercel. Local dev is fine. I have tried to swap them (because of the double whitespace in front of the output) but it is the same result.

Any idea what could cause the issue? Everything else is working.

The repo is private, but I could invite someone to investigate if necessary.

Tailwind version: 3.3.5
Both local and Vercel are using Node 18.

```ts
// tailwind.config.ts
import { join } from 'path'
import type { Config } from 'tailwindcss'
import forms from '@tailwindcss/forms';
import typography from '@tailwindcss/typography';
import { skeleton } from '@skeletonlabs/tw-plugin'
import { customTheme } from './custom-theme'

export default {
    darkMode: 'class',
    content: ['./src/**/*.{html,js,svelte,ts}', join(require.resolve('@skeletonlabs/skeleton'), '../**/*.{html,js,svelte,ts}')],
    theme: {
        extend: {},
    },
    plugins: [
        forms,
        typography,
        skeleton({
            themes: {
                custom: [
                    customTheme
                ],
            },
        }),
    ],
} satisfies Config;
```

**Comment by wongjn** [comment, marked as answer]
Tailwind needs to see classes as full text in files covered by the `content` file globs. With API data, that sounds like the class names would not be in your source code but instead produced by some server.

The `py-` classes work presumably because the associated CSS rules are generated from other source files in your project that use the exact same class. Conversely, the `bg-*` classes do not work since they do not appear anywhere in your source code.

To work around this, you can consider [safelisting](https://tailwindcss.com/docs/content-configuration#safelisting-classes) all classes that you anticipate will be retrieved from the API.

**Reply by carstenjaksch** [reply]
Thank you very much. Indeed, I use the `py` class in another component.

As a total Tailwind beginner, I have much to learn. But I like it alot. :)
